### PR TITLE
 Minor refactoring of and coverage bump for #36 

### PR DESCRIPTION
**Pull request by ayazhafiz**

### CODE REVIEW

File: src/Parser.hs [hunk 1]
```diff
@@ -5,15 +5,16 @@ module Parser
...
 import Error (Error(..), ErrorType(..), Expectation(..))
-import Grammar (Expression, Type, readType)
+import Grammar (Expression, Type)
 import qualified Grammar
 import Symbol (Symbol(..))
 import qualified Symbol
 import Token (Token(..))
 import Utils
 
+newtype Void = Void Void
```
Comment (lukasberglund) [inline src/Parser.hs]: Can you explain what's going on here? What do the two Voids refer to?

File: src/Parser.hs [hunk 2]
```diff
@@ -83,35 +82,33 @@ parseExpr precedence delims tokens =
 
 parseFunctionDeclaration :: ExpressionParser
 parseFunctionDeclaration tokens = do
-  expectedStart `assertMatches` tokens
+  tokens <- expectedStart `assertMatches` tokens
```
Comment (lukasberglund) [inline src/Parser.hs]: What's the point in renaming tokens here?

File: src/Parser.hs [hunk 3]
```diff
@@ -83,35 +82,33 @@ parseExpr precedence delims tokens =
...
   let nameTok:lParen:paramsAndBeyond = tokens
   (params, afterParams) <- parseDeclarations paramsAndBeyond
-  expectedMid `assertMatches` afterParams
+  tokens <- expectedMid `assertMatches` afterParams
```
Comment (lukasberglund) [inline src/Parser.hs]: And here

File: src/Parser.hs (hunk 1 above)
Comment (ayazhafiz) [reply src/Parser.hs]: sorry, i should remove this

File: src/Parser.hs (hunk 2 above)
Comment (ayazhafiz) [reply src/Parser.hs]: just to satisfy coverage, because `assertMatches` returns tokens on the right

**Pull request merged**